recipes/cuda-api-wrappers/all/conanfile.py

```diff
@@ -0,0 +1,38 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+class CudaKatConan(ConanFile):
+    name = "cuda-api-wrappers"
+    homepage = "https://github.com/eyalroz/cuda-api-wrappers"
+    description = "Thin C++-flavored wrappers for the CUDA APIs"
+    topics = ("gpgpu", "cuda", "cuda-api", "header-only")
+    url = "https://github.com/conan-io/conan-center-index"
+    no_copy_source = True
+    license = "BSD-3-Clause"
+
+    settings = "os", "arch", "compiler", "build_type"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def package_id(self):
+        self.info.header_only()
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("cuda-api-wrappers library are not compatible on Windows")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def package(self):
+        self.copy(pattern="LICENSE*", src=self._source_subfolder, dst="licenses")
+        self.copy("*", src=os.path.join(self._source_subfolder, "src"), dst="include")
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "cuda-api-wrappers"
+        self.cpp_info.names["cmake_find_package_multi"] = "cuda-api-wrappers"
```

Comment:
```suggestion
        # TODO: to remove in conan v2 once cmake_find_package* generators removed
        self.cpp_info.names["cmake_find_package"] = "cuda-api-wrappers"
        self.cpp_info.names["cmake_find_package_multi"] = "cuda-api-wrappers"  

        self.cpp_info.set_property("cmake_file_name", "cuda-api-wrappers")
        self.cpp_info.set_property("cmake_target_name", "cuda-api-wrapers::cuda-api-wrappers")
```
Better compatibility with Conan 2.0

Reply:
you can even remove `.names`

Reply:
Target name is wrong, there should be 3 imported targets.

Reply:
@SpaceIm Thank you!

Reply:
I'm a bit confused, how to declare 3 imported targets?

<details>
  <summary>Using components?</summary>
  
  ```
        # TODO: to remove in conan v2 once cmake_find_package* generators removed
        self.cpp_info.names["cmake_find_package"] = "cuda-api-wrappers"
        self.cpp_info.names["cmake_find_package_multi"] = "cuda-api-wrappers"

        self.cpp_info.set_property("cmake_file_name", "cuda-api-wrappers")
        self.cpp_info.components["runtime-and-driver"].set_property("cmake_target_name", "cuda-api-wrappers::runtime-and-driver")
        self.cpp_info.components["nvrtc"].set_property("cmake_target_name", "cuda-api-wrappers::nvrtc")
        self.cpp_info.components["nvtx"].set_property("cmake_target_name", "cuda-api-wrappers::nvtx")
```
</details>

Can you provide any example?